hathor/indexes/tips_index.py

```diff
@@ -48,8 +49,20 @@ class TipsIndex:
 
     def __init__(self) -> None:
         self.log = logger.new()
+        self.force_clear()
+
+    def get_db_name(self) -> Optional[str]:
+        return None
+
+    def force_clear(self) -> None:
         self.tree = IntervalTree()
-        self.tx_last_interval = {}  # Dict[bytes(hash), Interval]
+        self.tx_last_interval = {}
+
+    def init_loop_step(self, tx: BaseTransaction) -> None:
+        tx_meta = tx.get_metadata()
+        if not tx_meta.validation.is_final():
```

Comment:
Should we take this into consideration in other indexes?

Reply:
I didn't think too much about this, but possibly we should. I'll review this along with all the other validation related checks.

Reply:
Have you reviewed this?

Reply:
Yes, and I found out there's a problem: we're not guaranteed to visit not-fully-connected transactions, we will only visit them if they are partially linked through metadata when using the metadata topological iterator, and not at all when using the other iterators.

I think I prefer to address this in sync-v2.

Reply:
But this index is not used only by sync-v2, right? I feel that it is ok not to have transactions that are not connected to the genesis.

Reply:
You mean it's _not_ ok to have transactions that are not connected the to the genesis, right? This index definitely shouldn't have transactions that are not connected to the genesis. While it is used by the current sync-v2 implementation, it's use is small and will be easy to replace with either data from other indexes or a much simpler index.

But I don't think that is the point. Maybe we could temporarily guard all indexes from receiving transactions that aren't fully connected, or at least make a guard inside the iterator to make sure we're not visiting cases that aren't fully supported, having transactions that aren't fully connected isn't properly supported right now, and I think it will only make sense to fix this support on sync-v2, it's already hard enough to test this without actually using any of the sync-v2 code that actually make use of the stepped validation system.

Reply:
Agreed. Let's do it in the sync-v2 PR. Thanks!